### PR TITLE
x11-misc/xscreensaver: revbump, fix bugs

### PR DESCRIPTION
**Pull request by Schievel1**
Closes: https://bugs.gentoo.org/885441
Closes: https://bugs.gentoo.org/885479
Closes: https://bugs.gentoo.org/885989

Signed-off-by: Pascal Jäger <pascal.jaeger@leimstift.de>

### CODE REVIEW

File: x11-misc/xscreensaver/xscreensaver-6.05-r2.ebuild [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+	if ! use gtk; then
+		rm "${ED}/usr/bin/xscreensaver-demo" || die
+	fi
+	# Makefile installs xscreensaver.service regardless of --without-systemd
+	if ! use systemd; then
+		rm "${ED}/usr/share/${PN}/xscreensaver.service" || die
+	fi
+
+	#885989
+	fperms 4755 /usr/lib64/misc/xscreensaver/xscreensaver-auth
```
Comment (hartwork) [inline x11-misc/xscreensaver/xscreensaver-6.05-r2.ebuild:230]: Quick note that lib64 may not work for all users.

Comment (Flowdalic) [reply x11-misc/xscreensaver/xscreensaver-6.05-r2.ebuild:230]: Yes, you probably want to use `get_libdir` here
```suggestion
	fperms 4755 /usr/$(get_libdir)/misc/xscreensaver/xscreensaver-auth
```

File: x11-misc/xscreensaver/files/xscreensaver-6.05-r2-configure-exit-codes.patch [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+     warn2 "the \`webcollage' program will be much slower."
+-    CONF_STATUS=1
+   else
+     warn2 "This means the \`webcollage' program will be much slower."
+   fi
+@@ -4733,7 +4730,6 @@ if test "$have_png" = no ; then
+ 
+   if test "$png_halfassed" = yes ; then halfassery ; fi
+   warn2 "Many things aren't going to work right."
+-  CONF_STATUS=1
```
Comment (hartwork) [inline x11-misc/xscreensaver/files/xscreensaver-6.05-r2-configure-exit-codes.patch:40]: I commented on that approach at https://bugs.gentoo.org/885441#c3

File: x11-misc/xscreensaver/xscreensaver-6.05-r2.ebuild [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+		--with-xdbe-ext
+		--with-xf86gamma-ext
+		--with-xf86vmode-ext
+		--with-xinput-ext
+		--with-xkb-ext
+		--with-xshm-ext
+		--without-gle
+		--without-kerberos
+		--without-motif
+		--with-proc-oom=yes
```
Comment (hartwork) [inline x11-misc/xscreensaver/xscreensaver-6.05-r2.ebuild:183]: `-r2` seems to add `=yes` superflous `=yes` here. My vote for a revert, without it was cleaner. I checked both https://www.gnu.org/software/autoconf/manual/autoconf-2.60/html_node/External-Software.html and the `configure.ac` code.

**Pull request closed without merging**